File: apps/api-rs/src/api/history.rs
use axum::extract::{Query, State};
use axum::routing::get;
use axum::{Json, Router};
use serde::Deserialize;

use crate::auth::ActorExtractor;
use crate::models::conversation::ConversationListResponse;
use crate::state::AppState;

#[derive(Debug, Deserialize)]
pub struct HistoryQuery {
    #[serde(default = "default_limit")]
    limit: usize,
    #[serde(default)]
    cursor: Option<String>,
}

fn default_limit() -> usize {
    50
}

pub fn router() -> Router<AppState> {
    Router::new().route("/api/history", get(list_history))
}

async fn list_history(
    State(state): State<AppState>,
    ActorExtractor(actor): ActorExtractor,
    Query(query): Query<HistoryQuery>,
) -> Result<Json<ConversationListResponse>, crate::error::AppError> {
    let resp = state
        .repository
        .list_sessions(actor.tenant_id, actor.user_id, query.limit, query.cursor)
        .await?;
    Ok(Json(resp))
}

File: apps/api-bun/src/api/history.ts
// 移植自 apps/api-rs/src/api/history.rs
import { Hono } from 'hono';
import type { AppEnv } from '../http/types.ts';

export function historyRouter(): Hono<AppEnv> {
  const router = new Hono<AppEnv>();
  router.get('/api/history', async (c) => {
    const state = c.get('appState');
    const actor = c.get('actor');
    const limitRaw = c.req.query('limit');
    const limit = limitRaw !== undefined && limitRaw !== '' ? Number(limitRaw) : 50;
    const cursor = c.req.query('cursor') ?? null;
    const response = await state.repository.listSessions(
      actor.tenant_id, actor.user_id, Number.isFinite(limit) ? limit : 50, cursor);
    return c.json(response);
  });
  return router;
}
